### DISCUSSION TITLE
About the codomain of the control and admissibility

### DISCUSSION THREAD

**Discussion by ReHoss** [dynamicslab/hydrogym#91] [Q&A]
Hello:),

In general in finite-dimensional control theory the codomain of some control $u$ belongs to some subset $U \subset \mathbb{R}^m$.

In our case, how this subset is chosen ? For the flows involving cylinders, I guess it comes from the physical, real-life rotational settings.
What about the cavity or step flow for instance ? Indeed, I found in the source code a comment from @jcallaham which highlights this question:

https://github.com/dynamicslab/hydrogym/blob/b7174b2f039c968c61dbca8135bb1f15d5922098/hydrogym/firedrake/envs/step/flow.py#L17

Are there any references on how the actuation co-domain is defined for such blowing-suction configuration, like the reasoning behind the choice of the magnitude which then defines the above defined $U$ ? I see in the $\rho$ constant which defines the control magnitude is set to 1 in _A. Barbagallo, D. Sipp, P. J. Schmid - Closed-loop control of an open cavity flow using  reduced-order models (2014)_?

**Comment by jcallaham** [comment]
There's not an especially technical reason for these choices here.  If there's an experimental analogue, these should be tuned to match the physical actuators.  Here this constant is just to set limits that could be used by RL APIs ("box action spaces", for instance), and ideally to keep the controls within bounds that will be well-behaved numerically (though this hasn't been thoroughly tested).  But that comment is a reminder to look into whether or not those values are still used anywhere in the code.
